runtime/resolvers/legacy_metrics.go

```diff
@@ -0,0 +1,346 @@
+package resolvers
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"strconv"
+	"time"
+
+	"github.com/mitchellh/hashstructure/v2"
+	runtimev1 "github.com/rilldata/rill/proto/gen/rill/runtime/v1"
+	"github.com/rilldata/rill/runtime"
+	"github.com/rilldata/rill/runtime/pkg/formatter"
+	"github.com/rilldata/rill/runtime/pkg/mapstructureutil"
+	"github.com/rilldata/rill/runtime/queries"
+	"go.uber.org/zap"
+)
+
+func init() {
+	runtime.RegisterResolverInitializer("legacy_metrics", newLegacyMetrics)
+}
+
+type legacyMetricsResolver struct {
+	runtime         *runtime.Runtime
+	instanceID      string
+	query           runtime.Query
+	args            *legacyMetricsResolverArgs
+	metricsViewName string
+	logger          *zap.Logger
+}
+
+type legacyMetricsResolverProps struct {
+	QueryName     string `mapstructure:"query_name"`
+	QueryArgsJSON string `mapstructure:"query_args_json"`
+}
+
+type legacyMetricsResolverArgs struct {
+	Priority      int        `mapstructure:"priority"`
+	ExecutionTime *time.Time `mapstructure:"execution_time"`
+	Limit         int        `mapstructure:"limit"`
+	Format        bool       `mapstructure:"format"`
+}
+
+func newLegacyMetrics(ctx context.Context, opts *runtime.ResolverOptions) (runtime.Resolver, error) {
+	props := &legacyMetricsResolverProps{}
+	if err := mapstructureutil.WeakDecode(opts.Properties, props); err != nil {
+		return nil, err
+	}
+
+	args := &legacyMetricsResolverArgs{}
+	if err := mapstructureutil.WeakDecode(opts.Args, args); err != nil {
+		return nil, err
+	}
+
+	// Build query proto
+	qpb, err := queries.ProtoFromJSON(props.QueryName, props.QueryArgsJSON, args.ExecutionTime)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse query: %w", err)
+	}
+
+	metricsViewName, err := queries.MetricsViewFromQuery(props.QueryName, props.QueryArgsJSON)
+	if err != nil {
+		return nil, fmt.Errorf("failed extract metrics view name from query: %w", err)
+	}
+
+	q, err := queries.ProtoToQuery(qpb, opts.UserAttributes)
```

Comment:
After the introduction of shareable URLs recently, this now also needs to incorporate `opts.SecurityPolicy` (a security policy provided in the JWT instead of in the resource itself)